packages/monitor-v2/src/monitor-og/common.ts

```diff
@@ -129,6 +138,47 @@ export const initMonitoringParams = async (env: NodeJS.ProcessEnv): Promise<Moni
     pollingDelay,
     botModes,
   };
+
+  // If OG_ADDRESS is provided, use it in the monitored address list and return monitoring params.
+  // Invalid address will throw an error in getAddress call.
+  if (env.OG_ADDRESS) {
+    initialParams.ogAddresses = [utils.getAddress(env.OG_ADDRESS)];
+    return initialParams;
+  }
+
+  // Verify that OG whitelist and blacklist contain only deployed proxy addresses.
+  // Invalid addresses will throw an error in getAddress call.
+  const deployedProxyAddresses = await getDeployedProxyAddresses(initialParams, {
+    start: 0,
+    end: initialParams.blockRange.end,
+  });
+  const ogWhitelist: string[] = env.OG_WHITELIST ? JSON.parse(env.OG_WHITELIST) : [];
+  const ogBlacklist: string[] = env.OG_BLACKLIST ? JSON.parse(env.OG_BLACKLIST) : [];
+  ogWhitelist.forEach((address) => {
+    if (!deployedProxyAddresses.includes(utils.getAddress(address))) {
+      throw new Error(`OG_WHITELIST contains address ${address} that is not a deployed proxy`);
+    }
+  });
+  ogBlacklist.forEach((address) => {
+    if (!deployedProxyAddresses.includes(utils.getAddress(address))) {
+      throw new Error(`OG_BLACKLIST contains address ${address} that is not a deployed proxy`);
+    }
+  });
+
+  // If OG whitelist is provided, use it in the monitored address list and return monitoring params.
+  if (env.OG_WHITELIST) {
+    initialParams.ogAddresses = ogWhitelist.map((address) => utils.getAddress(address));
+    return initialParams;
+  }
+
+  // We are in automatic OG address discovery mode. Return monitoring params with all deployed proxies except those
+  // in the blacklist.
+  initialParams.ogAddresses = deployedProxyAddresses.filter(
+    (address) => !ogBlacklist.includes(utils.getAddress(address))
+  );
+  initialParams.ogBlacklist = ogBlacklist.map((address) => utils.getAddress(address));
```

Comment:
nit: have you considered putting this logic into getOgAddresses?

For instance:
1. Just set ogWhitelist and ogBlacklist in the params. Everything else is dynamic.
2. Put all the logic to grab all deployed proxies and verify ogWhitelist and ogBlacklist in that function (this would be run on each loop unnecessarily, but it shouldn't be super expensive).
3. Run getOgAddresses at the beginning of each loop.

Thoughts? Do you think that would be simpler (maybe not!).

Reply:
yes, have considered this initially, but did not as:
- this adds get logs calls from block 0 on each loop. its not a real difference for serverless, but might be impactful for external teams running the bot in looping mode
- this required storing OG_ADDRESS in params (in addition to whitelist/blacklist) as there might be use cases to monitor specific OG_ADDRESS that is not deployed through proxy
- it seemed better keep all env verification logic in the initMonitoringParams function.